rcsb/app/client/python/ClientUtils.py

```diff
@@ -297,31 +353,31 @@ def download(
         partNumber: int,
         contentFormat: str,
         version: str,
-        hashType: str,
-        downloadFolder: str,
-        allowOverwrite: bool
+        hashType: str = "MD5",
+        downloadFolder: typing.Optional[str] = None,
+        allowOverwrite: bool = False,
+        returnTempFile: bool = False,
+        deleteTempFile: bool = True
     ):
-        if not os.path.exists(downloadFolder):
-            logger.error("Download folder does not exist")
-            return None
         convertedMilestone = None
+        if not milestone or milestone.lower() == "none":
+            milestone = ""
         if milestone and milestone.lower() != "none":
             convertedMilestone = f"-{milestone}"
         else:
             convertedMilestone = ""
         convertedContentFormat = self.fileFormatExtensionD[contentFormat]
```

Comment:
Can you explain what the purpose of this is? Why must you convert the milestone? What exactly are you doing to it?